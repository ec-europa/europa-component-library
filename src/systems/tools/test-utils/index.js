const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
