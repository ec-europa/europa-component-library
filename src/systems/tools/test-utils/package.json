{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
