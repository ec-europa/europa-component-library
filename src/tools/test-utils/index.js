const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
