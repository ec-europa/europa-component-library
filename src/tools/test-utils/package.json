{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
