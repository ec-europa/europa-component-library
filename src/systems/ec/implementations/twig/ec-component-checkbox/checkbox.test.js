import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import dataDefault from './demo/data--default';
import dataInvalid from './demo/data--invalid';

describe('EC - checkbox', () => {
  const template =
    '@ecl-twig/ec-component-checkbox/ecl-checkbox-group.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test('renders correctly when required', () => {
      expect.assertions(1);

      dataDefault.items.forEach((item) => {
        item.required = true; // eslint-disable-line no-param-reassign
      });

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });
  });

  describe('Invalid', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataInvalid)).resolves.toMatchSnapshot();
    });
  });

  describe('with validation enabled and missing input data', () => {
    test('returns the right warning message', () => {
      expect.assertions(1);
      const dataCompliance = { ...dataDefault, _compliance_: true };
      dataCompliance.items[0].label = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
