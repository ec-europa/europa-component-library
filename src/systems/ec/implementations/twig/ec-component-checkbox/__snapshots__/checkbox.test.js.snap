// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - checkbox Default renders correctly 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group"
  >
    <legend
      class="ecl-form-label"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-default-1"
        name="checkbox-default"
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-1"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Spain
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-1"
      >
        Help text for option 1
      </div>
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-default-2"
        name="checkbox-default"
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-2"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-2"
      >
        Help text for option 2
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-default-3"
        name="checkbox-default"
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-default-3"
      >
        Help text for option 3
      </div>
    </div>
  </fieldset>
</test>
`;

exports[`EC - checkbox Default renders correctly when required 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group"
  >
    <legend
      class="ecl-form-label"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-default-1"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-1"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Spain
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-1"
      >
        Help text for option 1
      </div>
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-default-2"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-2"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-2"
      >
        Help text for option 2
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-default-3"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-default-3"
      >
        Help text for option 3
      </div>
    </div>
  </fieldset>
</test>
`;

exports[`EC - checkbox Default renders correctly with extra attributes 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <legend
      class="ecl-form-label"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-default-1"
        name="checkbox-default"
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-1"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Spain
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-1"
      >
        Help text for option 1
      </div>
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-default-2"
        name="checkbox-default"
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-2"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-2"
      >
        Help text for option 2
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-default-3"
        name="checkbox-default"
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-default-3"
      >
        Help text for option 3
      </div>
    </div>
  </fieldset>
</test>
`;

exports[`EC - checkbox Default renders correctly with extra class names 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group custom-class custom-class--test"
  >
    <legend
      class="ecl-form-label"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-default-1"
        name="checkbox-default"
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-1"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Spain
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-1"
      >
        Help text for option 1
      </div>
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-default-2"
        name="checkbox-default"
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-2"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-2"
      >
        Help text for option 2
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-default-3"
        name="checkbox-default"
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-default-3"
      >
        Help text for option 3
      </div>
    </div>
  </fieldset>
</test>
`;

exports[`EC - checkbox Invalid renders correctly 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group"
  >
    <legend
      class="ecl-form-label ecl-form-label--invalid"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-feedback-message"
    >
      Error message for the group
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--invalid"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-invalid-1"
        name="checkbox-invalid"
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-invalid-1"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--invalid"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Spain
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-invalid-1"
      >
        Help text for option 1
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--invalid"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-invalid-2"
        name="checkbox-invalid"
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-invalid-2"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--invalid"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-invalid-2"
      >
        Help text for option 2
      </div>
    </div>
    <div
      class="ecl-checkbox ecl-checkbox--invalid ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-invalid-3"
        name="checkbox-invalid"
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-invalid-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--invalid ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-invalid-3"
      >
        Help text for option 3
      </div>
    </div>
  </fieldset>
</test>
`;

exports[`EC - checkbox with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <fieldset
    aria-describedby="checkbox-default-helper"
    class="ecl-form-group"
  >
    <legend
      class="ecl-form-label"
      id="checkbox-default-label"
    >
      Select your preferred destinations
      <span
        class="ecl-form-label__optional"
      >
         (optional)
      </span>
    </legend>
    <div
      class="ecl-help-block"
      id="checkbox-default-helper"
    >
      Helper text for the group
    </div>
    <div
      class="ecl-checkbox"
    >
      <input
        checked=""
        class="ecl-checkbox__input"
        id="checkbox-default-1"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="es"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-1"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </span>
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-1"
      >
        Help text for option 1
      </div>
    </div>
    <span
      class="ecl-u-type-color-red ecl-compliance-inner ecl-compliance-failure ecl-u-ma-2xs"
      is=""
      label"=""
      mandatory."=""
      title="[checkbox] - The "
    >
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--xs"
        focusable="false"
      >
        <use
          xlink:href="/icons.svg#notifications--error"
        />
      </svg>
    </span>
    <div
      class="ecl-checkbox"
    >
      <input
        class="ecl-checkbox__input"
        id="checkbox-default-2"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="be"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-2"
      >
        <span
          class="ecl-checkbox__box"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </span>
        Belgium
      </label>
      <div
        class="ecl-checkbox__help"
        id="helper-default-2"
      >
        Help text for option 2
      </div>
    </div>
    <span
      class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
      title="[checkbox] - ecl compliant"
    >
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--xs"
        focusable="false"
      >
        <use
          xlink:href="/icons.svg#notifications--success"
        />
      </svg>
    </span>
    <div
      class="ecl-checkbox ecl-checkbox--disabled"
    >
      <input
        class="ecl-checkbox__input"
        disabled=""
        id="checkbox-default-3"
        name="checkbox-default"
        required=""
        type="checkbox"
        value="fr"
      />
      <label
        class="ecl-checkbox__label"
        for="checkbox-default-3"
      >
        <span
          class="ecl-checkbox__box ecl-checkbox__box--disabled"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--s ecl-checkbox__icon"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#ui--check"
            />
          </svg>
          <span
            class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
            title="[icon] - ecl compliant"
          >
            <svg
              aria-hidden="true"
              class="ecl-icon ecl-icon--xs"
              focusable="false"
            >
              <use
                xlink:href="/icons.svg#notifications--success"
              />
            </svg>
          </span>
        </span>
        France (disabled)
      </label>
      <div
        class="ecl-checkbox__help ecl-checkbox__help--disabled"
        id="helper-default-3"
      >
        Help text for option 3
      </div>
    </div>
    <span
      class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
      title="[checkbox] - ecl compliant"
    >
      <svg
        aria-hidden="true"
        class="ecl-icon ecl-icon--xs"
        focusable="false"
      >
        <use
          xlink:href="/icons.svg#notifications--success"
        />
      </svg>
    </span>
  </fieldset>
  <!-- ecl2-compliance: form-group--checkbox -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-green ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-s ecl-u-type-xs"
    >
      [form-group--checkbox] - ecl compliant
    </p>
  </div>
</test>
`;
